transcode/transcode.go

```diff
@@ -165,6 +156,92 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 		return outputs, segmentsCount, err
 	}
 
+	// transmux received segments from T into a single mp4
+	if transcodeRequest.GenerateMP4 {
+		for rlist, slist := range renditionList.RenditionSegmentTable {
+			// create folder
+			if _, err := os.Stat(TRANSMUX_STORAGE_DIR); os.IsNotExist(err) {
+				os.MkdirAll(TRANSMUX_STORAGE_DIR, 0700)
+			}
+
+			// 1. create a .ts file for a given rendition
+			fileName := filepath.Join(TRANSMUX_STORAGE_DIR, transcodeRequest.RequestID+"_"+rlist)
+			f, err := os.Create(fileName+".ts")
+			if err != nil {
+				log.Log(transcodeRequest.RequestID, "XXX: error creating file", fileName, "err", err)
+				f.Close()
+				continue
+			}
+			defer f.Close()
+
+			// 2. for a given rendition, write all segment indices in ascending order to the single .ts file
+			var totalBytes int64
+			for _, k := range slist.GetSortedSegments() {
+//				log.Log(transcodeRequest.RequestID, "XXX:", k, len(slist.SegmentDataTable[k]))
+				fmt.Println(transcodeRequest.RequestID, "XXX:", k, len(slist.SegmentDataTable[k]))
+				segBytes, err := f.Write(slist.SegmentDataTable[k])
+				if err != nil {
+					log.Log(transcodeRequest.RequestID, "XXX: error writing segment", k, "err", err)
+					f.Close()
+					break
+				}
+				totalBytes = totalBytes + int64(segBytes)
+			}
+			// Verify the total bytes written for a given rendition matches the total # of bytes we received from T
+			renditionIndex := getProfileIndex(transcodeProfiles, rlist)
+			fmt.Println("XXX: ", renditionIndex, "transcodeProfiles", transcodeProfiles, "rlist", rlist, "transcodedStats", transcodedStats)
+			var rendBytesWritten int64 = -1
+			for _, v := range transcodedStats {
+				if v.Name == rlist {
+					rendBytesWritten = v.Bytes
```

Comment:
it doesn't look like this `Bytes` value will be filled in.. the stats are populated here: https://github.com/livepeer/catalyst-api/blob/ae7536c9a4ff9f95db94646ab83208af69cf9908/transcode/transcode.go#L255

Reply:
Actually comes from here: 
https://github.com/livepeer/catalyst-api/blob/b003b72d49822911cd0df143fcb8a86cb6e6c650/transcode/transcode.go#L251

And during testing, I can see it get populated as expected as we get results from transcoder.